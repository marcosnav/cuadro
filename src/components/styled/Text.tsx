import styled from 'styled-components';

interface IProps {
  bottom?: boolean;
}

const Text = styled.p<IProps>`
  bottom: ${({ bottom }) => bottom ? '16px' : 'auto'};
  color: ${(props) => props.theme.TEXT};
  font-size: 0.7em;
  left: ${({ bottom }) => bottom ? '0' : 'auto'};
  line-height: 2.97em;
  margin: auto;
  max-width: 420px;
  position: ${({ bottom }) => bottom ? 'fixed' : 'relative'};
  right: ${({ bottom }) => bottom ? '0' : 'auto'};
  text-align: center;
  width: 100%;
`;

Text.displayName = 'Text';

export default Text;
