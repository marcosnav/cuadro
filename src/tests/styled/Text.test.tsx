import { mount } from 'enzyme';
import 'jest-styled-components';
import React from 'react';
import Text from './../../components/styled/Text';

describe('<Text/>', () => {
  test('renders properly', () => {
    const component = mount(<Text bottom={true} >Made by john</Text>);
    expect(component.text()).toMatch('Made by john');
    expect(component).toMatchSnapshot();
  });

  test('bottom: false', () => {
    const component = mount(<Text bottom={false} >Made by john</Text>);
    expect(component).toHaveStyleRule('bottom', 'auto');
    expect(component).toHaveStyleRule('left', 'auto');
    expect(component).toHaveStyleRule('position', 'relative');
  });
});
