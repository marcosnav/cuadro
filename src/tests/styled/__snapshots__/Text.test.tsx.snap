// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text/> renders properly 1`] = `
.c0 {
  bottom: 16px;
  font-size: 0.7em;
  left: 0;
  line-height: 2.97em;
  margin: auto;
  max-width: 420px;
  position: fixed;
  right: 0;
  text-align: center;
  width: 100%;
}

<Text
  bottom={true}
>
  <p
    className="c0"
  >
    Made by john
  </p>
</Text>
`;
